types/events.go

```diff
@@ -62,24 +64,24 @@ func init() {
 // but some (an input to a call tx or a receive) are more exotic
 
 type EventDataNewBlock struct {
-	Block *Block `json:"block"`
-
-	ResultBeginBlock abci.ResponseBeginBlock `json:"result_begin_block"`
-	ResultEndBlock   abci.ResponseEndBlock   `json:"result_end_block"`
+	Block               *Block                     `json:"block"`
+	BlockID             BlockID                    `json:"block_id"`
+	ResultFinalizeBlock abci.ResponseFinalizeBlock `json:"result_finalize_block"`
```

Comment:
> There's a lot more being stored here than previously.

`EventDataNewBlock` isn't used by anything internal to tendermint (i.e. not by the block and tx indexer). I've just added `ResultFinalizeBlock` here to somewhat mimic what was previously there but we could also remove it so that `EventDataNewBlock` just provides the block. This does mean there is no event for consensus param changes which some users may want to track.

I actually created `EventDataNewBlockEvents` specifically because we were piping out an entire block to the block indexer when just the events were necessary. 

Just to recap, the block indexer uses `EventDataNewBlockEvents` and the tx indexer takes in `EventDataTx`

Reply:
Just a quick note that I just showed this part to @jmalicevic in the context of the indexing problem she's tackling and she finds it cleaner than the status quo (from event indexing perspective)

Reply:
> I actually created EventDataNewBlockEvents specifically because we were piping out an entire block to the block indexer when just the events were necessary.

I guess I'm unclear on what the issue with this is. We're presenting the exact same data in two formats: `EventDataNewBlockEvents` contains no data that isn't contained in `EventDataNewBlock`.

Reply:
`EventDataNewBlock` contains everything though. Every event could be contained within `EventDataNewBlock`. It's existence serves as a catch all for external clients. Nowhere in Tendermint do we actually use it. Since we have a block indexer and a tx indexer, I'd prefer that the events are just scoped to what these two indexers need: tx and tx events, and block events.